### PR TITLE
 ⚠️ Accomodate CAPI v1a4 related changes

### PR DESCRIPTION
**Pull request by kashifest**
This PR accommodates infra related CAPI v1a4 related changes as described `[here](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/book/src/developer/providers/v1alpha3-to-v1alpha4.md)`

### CODE REVIEW

**Review by lentzi90 — APPROVED** [review]

**Review by macaptain — APPROVED** [review]
Looks good to me.

File: main.go [hunk 1]
```diff
@@ -38,32 +41,47 @@ import (
...
 	_ = clusterv1.AddToScheme(myscheme)
+	_ = clusterv1a3.AddToScheme(myscheme)
 	// +kubebuilder:scaffold:scheme
 }
 
 func main() {
 	klog.InitFlags(nil)
-	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
+	flag.StringVar(&metricsBindAddr, "metrics-bind-addr", "localhost:8080", "The address the metric endpoint binds to.")
 	flag.BoolVar(&enableLeaderElection, "enable-leader-election", false,
```
Comment (furkatgofurov7) [inline main.go:65]:
```suggestion
	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
```

**Review by furkatgofurov7 — CHANGES_REQUESTED** [review]

**Review by furkatgofurov7 — COMMENTED** [review]
Thanks for taking care of it, good job! I have some questions and suggestions inline:
- We could update [controller tools ](https://github.com/metal3-io/ip-address-manager/blob/master/hack/tools/go.mod#L9) as per infra related [changes](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/book/src/developer/providers/v1alpha3-to-v1alpha4.md#controller-tools-version-if-used) as we do in CAPM3 PR

File: main.go [hunk 2]
```diff
@@ -38,32 +41,47 @@ import (
...
 	watchNamespace       string
+	webhookCertDir       string
+	watchFilterValue     string
 )
 
 func init() {
 	_ = scheme.AddToScheme(myscheme)
 	_ = ipamv1.AddToScheme(myscheme)
 	_ = clusterv1.AddToScheme(myscheme)
+	_ = clusterv1a3.AddToScheme(myscheme)
```
[the remark this answers is not in the document]
Comment (kashifest) [reply main.go:58]: We need this for backword compatibility, cluster v1a3 is capi v1a3.

**Review by furkatgofurov7 — APPROVED** [review]
/approve

File: main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (kashifest) [reply main.go]: I think this is not needed since no other provider is using it so removing it.

File: main.go [hunk 3]
```diff
@@ -55,15 +59,27 @@ func init() {
...
+	flag.StringVar(&metricsBindAddr, "metrics-bind-addr", "localhost:8080", "The address the metric endpoint binds to.")
+	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
 		"Enable leader election for controller manager. Enabling this will ensure there is only one active controller manager.")
 	flag.StringVar(&watchNamespace, "namespace", "",
 		"Namespace that the controller watches to reconcile CAPM3 objects. If unspecified, the controller watches for CAPM3 objects across all namespaces.")
 	flag.DurationVar(&syncPeriod, "sync-period", 10*time.Minute,
 		"The minimum interval at which watched resources are reconciled (e.g. 15m)")
-	flag.IntVar(&webhookPort, "webhook-port", 0,
+	flag.IntVar(&webhookPort, "webhook-port", 9443,
 		"Webhook Server port (set to 0 to disable)")
```
Comment (wgslr) [inline main.go:70]: Just checking: is `set to 0 to disable` still true, with the checks `webhookPort == 0` removed in other files?

Comment (furkatgofurov7) [reply main.go]: Fixed, thank you.

**Comment by furkatgofurov7** [comment]
This needs a force merge since it can not pass integration tests until m3-dev-env [PR](https://github.com/metal3-io/metal3-dev-env/pull/693)

**Comment by fmuyassarov** [comment]
merging it manually to get the CI back and take these changes

**Pull request merged**